packages/jest-mock/src/index.ts

```diff
@@ -1230,6 +1273,117 @@ export class ModuleMocker {
     return descriptor[accessType] as Mock;
   }
 
+  replaceProperty<
+    T extends object,
+    K extends PropertyLikeKeys<T>,
+    V extends T[K],
+  >(object: T, propertyKey: K, value: V): Replaced<T[K]> {
+    if (object === undefined || object == null) {
+      throw new Error(
+        `replaceProperty could not find an object on which to replace ${String(
+          propertyKey,
+        )}`,
+      );
+    }
+
+    if (propertyKey === undefined || propertyKey === null) {
+      throw new Error('No property name supplied');
+    }
+
+    if (typeof object !== 'object') {
+      throw new Error(
+        `Cannot mock property on a non-object value; ${this._typeOf(
+          object,
+        )} given`,
+      );
+    }
+
+    let descriptor = Object.getOwnPropertyDescriptor(object, propertyKey);
+    let proto = Object.getPrototypeOf(object);
+    while (!descriptor && proto !== null) {
+      descriptor = Object.getOwnPropertyDescriptor(proto, propertyKey);
+      proto = Object.getPrototypeOf(proto);
+    }
+    if (!descriptor) {
+      throw new Error(`${String(propertyKey)} property does not exist`);
+    }
+    if (!descriptor.configurable) {
+      throw new Error(`${String(propertyKey)} is not declared configurable`);
+    }
+
+    if (descriptor.get !== undefined) {
+      throw new Error(
+        `Cannot mock the ${String(
+          propertyKey,
+        )} property because it has a getter. Use \`jest.spyOn(object, '${String(
```

Comment:
I checked it message you're probably referring to (`'spyOn' is called on primitive value`) is thrown when first argument of `spyOn()` [is invalid](https://github.com/facebook/jest/blob/cceffa0e9689fe73b667d19e1b84a151ab83250f/packages/jest-mock/src/index.ts#L1140-L1144).

The guidance might make sense when user tries to mock property which is [not a function](https://github.com/facebook/jest/blob/cceffa0e9689fe73b667d19e1b84a151ab83250f/packages/jest-mock/src/index.ts#L1162-L1171). However... In this case, the message will be thrown in  other cases as well - for example when value is null, undefined, when method does not exist on the object... I reviewed [recent StackOverflow searches](https://stackoverflow.com/search?tab=newest&q=%22property%20because%20it%20is%20not%20a%20function%22) of that error message. It seems that majority of people are trying to spy on methods that do not exist for various reasons, not because they'd want to replace a property... In which case my personal feeling is that the guidance could be more misleading, as it might cause more false positives... (Unless we'd specifically check for - the property has a value, not undefined, not null, and then maybe raised the message - in which case it would be super niche case (still just my subjective assumption)).

Will appreciate your thoughts on this...

Reply:
What about keep the current message, but adding something like: "If you are trying to mock a property, use `jest. replaceProperty()` instead."

Reply:
Yeah, adding that as an extra hint to the user sounds good to me 👍